app/src/ui/add-repository/add-existing-repository.tsx

```diff
@@ -81,18 +103,39 @@ export class AddExistingRepository extends React.Component<IAddExistingRepositor
     this.checkIfPathIsRepository(path)
   }
 
+  private pathExists(directory: string): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      try {
+        FS.stat(directory, (err, exists) => {
+          if (err) {
+            resolve(false)
+          }
+
+          resolve(true)
+        })
+      } catch (e) {
+        resolve(false)
+      }
+    })
+  }
+
   private async checkIfPathIsRepository(path: string) {
-    this.setState({ path, isGitRepository: null })
+    this.setState({ path })
 
     const token = ++this.checkGitRepositoryToken
+    const pathExists = await this.pathExists(path)
```

Comment:
This is subject to a race condition, albeit a narrow one (but then again, the best ones are).

After thinking about this some more I think the best course of action here is to catch the specific error message that dugite throws when a repository can't be found (RepositoryDoesNotExistErrorCode). And I think we should do so inside `getGitDir` itself (or rather `getTopLevelWorkingDirectory` once master has been merged into this branch). That way we can avoid this `stat` and be sure there's no race condition.